contrib/runners/action_chain_runner/tests/unit/test_actionchain.py

```diff
@@ -464,14 +466,22 @@ def test_chain_runner_dependent_results_param(self, request):
         chain_runner.pre_run()
         chain_runner.run({'s1': 1})
         self.assertNotEqual(chain_runner.chain_holder.actionchain, None)
-        expected_values = [{u'p1': u'1'},
-                           {u'p1': u'1'},
-                           {u'out': u"{'c2': {'o1': '1'}, 'c1': {'o1': '1'}}"}]
+
+        if six.PY2:
+            expected_values = [{u'p1': u'1'},
```

Comment:
One option would be to modify how the `assertEquals`, etc. works because the comparison fails right now.

One of the problems also is that in a lot of cases we are comparing a string and not a native type and in that case, `u` also causes problems.

And I do agree that this approach is not the best :)